### ISSUE TITLE
Bug: Eval panics.

### ISSUE THREAD

**Issue by james-lawrence** [containous/yaegi#413]
it shouldn't. it already returns an error. it should recover itself and return that error.
```go
// Eval evaluates Go code represented as a string. It returns a map on
// current interpreted package exported symbols
func (interp *Interpreter) Eval(src string) (res reflect.Value, err error) {
	defer func() {
		recovered := recover()
		if recovered == nil {
			return
		}

		switch cause := recovered.(type) {
		case runtime.Error:
                        panic(cause)
		case error:
                        err = cause
		}
	}()

	// Parse source to AST
	pkgName, root, err := interp.ast(src, interp.Name)
	if err != nil || root == nil {
		return res, err
	}

	if interp.astDot {
		root.astDot(dotX(), interp.Name)
		if interp.noRun {
			return res, err
		}
	}

	// Global type analysis
	revisit, err := interp.gta(root, pkgName)
	if err != nil {
		return res, err
	}
	for _, n := range revisit {
		if _, err = interp.gta(n, pkgName); err != nil {
			return res, err
		}
	}

	// Annotate AST with CFG infos
	initNodes, err := interp.cfg(root)
	if err != nil {
		return res, err
	}

	// Add main to list of functions to run, after all inits
	if m := interp.main(); m != nil {
		initNodes = append(initNodes, m)
	}

	if root.kind != fileStmt {
		// REPL may skip package statement
		setExec(root.start)
	}
	if interp.universe.sym[pkgName] == nil {
		// Make the package visible under a path identical to its name
		interp.srcPkg[pkgName] = interp.scopes[pkgName].sym
		interp.universe.sym[pkgName] = &symbol{kind: pkgSym, typ: &itype{cat: srcPkgT, path: pkgName}}
	}

	if interp.cfgDot {
		root.cfgDot(dotX())
	}

	if interp.noRun {
		return res, err
	}

	// Execute CFG
	if err = genRun(root); err != nil {
		return res, err
	}
	interp.resizeFrame()
	interp.run(root, nil)

	for _, n := range initNodes {
		interp.run(n, interp.frame)
	}
	v := genValue(root)
	res = v(interp.frame)

	// If result is an interpreter node, wrap it in a runtime callable function
	if res.IsValid() {
		if n, ok := res.Interface().(*node); ok {
			res = genFunctionWrapper(n)(interp.frame)
		}
	}

	return res, err
}
```

**Comment by mvertes** [comment]
Yes, you're right. We should just return an error, possibly containing the stack trace of a recovered panic.

**Comment by kortschak** [comment]
@mvertes I have a branch that does this that I'm happy to send. Note that the stack trace is non-trivial since there are two stacks that may be relevant: 1. the Go runtime stack and 2. the yaegi stack. I think the first step is just to put in the error return and then sort out the stack traces later.

**Comment by mvertes** [comment]
Fixed by #508
